Marlin/Marlin_main.cpp

```diff
@@ -1487,6 +1492,7 @@ bool get_target_extruder_from_command(int code) {
   static void set_home_offset(const AxisEnum axis, const float v) {
     current_position[axis] += v - home_offset[axis];
     home_offset[axis] = v;
+    update_workspace_offset(axis);
     update_software_endstops(axis);
```

Comment:
If every single call to `update_software_endstops(axis)` is being coupled with a preceding call to `update_workspace_offset(axis)` then the functions should simply be combined.

Note: There is also a call to `update_software_endstops(axis)` in `configuration_store.cpp`.